eval/eval_adapter.py

```diff
@@ -0,0 +1,362 @@
+
+    return idx
+
+
+class EvalHarnessAdapter(BaseLM):
+    # Credits:
+    # https://github.com/EleutherAI/gpt-neox/blob/main/eval_tasks/eval_adapter.py
+    def __init__(
+        self,
+        checkpoint_dir: str = "",
+        precision: str = "float32",
+        batch_size=1,
+        temperature=1.0,
+        device="auto",
+        devices: int = 1,
+        strategy: str = "auto",
+        quantize: Optional[
+            Literal[
+                "bnb.nf4",
+                "bnb.nf4-dq",
+                "bnb.fp4",
+                "bnb.fp4-dq",
+                "bnb.int8",
+                "gptq.int4",
+            ]
+        ] = None,
+    ):
+        super().__init__()
+        assert isinstance(device, str)
+        assert isinstance(batch_size, int)
+        assert isinstance(checkpoint_dir, str)
+
+        if strategy == "fsdp":
+            strategy = FSDPStrategy(auto_wrap_policy={Block}, cpu_offload=False)
+        self.fabric = fabric = L.Fabric(
+            devices=devices, precision=precision, strategy=strategy
+        )
+        fabric.launch()
+
+        checkpoint_dir = Path(checkpoint_dir)
+
+        check_valid_checkpoint_dir(checkpoint_dir)
+
+        with open(checkpoint_dir / "lit_config.json") as fp:
+            config = Config(**json.load(fp))
+
+        if quantize is not None and devices > 1:
+            raise NotImplementedError
+        if quantize == "gptq.int4":
+            model_file = "lit_model_gptq.4bit.pth"
+            if not (checkpoint_dir / model_file).is_file():
+                raise ValueError("Please run `python quantize/gptq.py` first")
+        else:
+            model_file = "lit_model.pth"
+        checkpoint_path = checkpoint_dir / model_file
+
+        fabric.print(
+            f"Loading model {str(checkpoint_path)!r} with {config.__dict__}",
+            file=sys.stderr,
+        )
+        t0 = time.time()
+        with fabric.init_module(empty_init=True), quantization(quantize):
+            model = GPT(config)
+        fabric.print(
+            f"Time to instantiate model: {time.time() - t0:.02f} seconds.",
+            file=sys.stderr,
+        )
+
+        t0 = time.time()
+        with lazy_load(checkpoint_path) as checkpoint:
+            model.load_state_dict(
+                checkpoint.get("model", checkpoint), strict=quantize is None
+            )
+        fabric.print(
+            f"Time to load the model weights: {time.time() - t0:.02f} seconds.",
+            file=sys.stderr,
+        )
+
+        model.eval()
+        self.model = fabric.setup_module(model)
+        self.tokenizer = Tokenizer(checkpoint_dir)
+        self.vocab_size = self.tokenizer.vocab_size
+
+        # multithreading and batching
+        self.batch_size_per_gpu = batch_size
+
+        self.temperature = temperature
+
+    @classmethod
+    def create_from_arg_string(cls, arg_string, additional_config=None):
+        kwargs = {el.split("=")[0]: el.split("=")[1] for el in arg_string.split(",")}
```

Comment:
Interestingly enough this one should work (though I haven't run your code 🙃 ):
```python
kwargs = dict(el.split("=") for el in arg_string.split(","))
```

Reply:
Plus it's a bit more performant as there is no need to split each `el` twice. 
Python go brrrr!

<img width="750" alt="Screenshot 2023-08-06 at 6 53 32 PM" src="https://github.com/Lightning-AI/lit-gpt/assets/58434077/a9e15c5f-7b1b-4464-b2a1-6080c5cf099e">

Just a Sunday joke 😄

Reply:
nice find! I will do some cleanup in the PR tomorrow morning 😄